dateutil/parser/_parser.py

```diff
@@ -1464,6 +1492,8 @@ def parse(self, tzstr):
         except (IndexError, ValueError, AssertionError):
             return None
 
+        unused_idxs = set(range(len_l)).difference(used_idxs)
+        res.any_unused_tokens = not {l[n] for n in unused_idxs}.issubset({",",":"})
```

Comment:
It's kinda amazing to me that this seems to be *universally* faster than `any(l[n] not in {',', ':'} for n in unused_idxs)`, even when you pull the `{',', ':'}` creation out of the loop.

Maybe if you have some absurd number of tokens the set creation portion would get faster and this would get slow.